### PR TITLE
React Week 1 Changes

### PR DESCRIPTION
**Pull request by droberts-sea**
## Description
Start with functional components. Includes a large number of small clarifications and updates.

## Questions for the Author
- [x] Have you updated the weekly learning goals in the pedagogy resource?
- [x] Have you added links to the source (lucidchart, draw.io) for any diagram?
- [x] Have you updated the READMEs with links to new or moved resources?

## Todos
- [ ] End-of-week project ([Radio Lovelace](https://github.com/AdaGold/radio-lovelace))
- [x] Intro to classes #482

## Feedback Requested
Probably the best approach would be to follow the order in the README.

I've tested all the code myself, so I'm mostly interested in feedback on prose and whether or not we do a good job of hitting the learning goals.

Also, I will be OOO starting Monday 11/26, so if you see something that needs fixing, fix it!

### CODE REVIEW

File: React/nested-event-handling.md [hunk 1]
```diff
@@ -1,66 +1,89 @@
...
+## Refactor to a Stateful Component
 
-1. Spend a few minutes now going back to the [creating components](creating-components.md) notes to see how to create the new component file with the basic class, imports and exports you need.
+### Lifting State Up
 
-1. Move the code that renders the list of `Student`s from `App` to `StudentCollection`. Note: You will also need to move and update the `import` statement.
+Currently there is no way for us to change the student data. If we want to be able to change whether a student is present, we'll need to keep track of that as state in our application.
 
-1. Import and render `StudentCollection` now from the `App` component instead. Note: You will need to add an import statement.
+**Question:** Where should our application's state live? What are our options?
```
Comment (tildeee) [inline React/nested-event-handling.md]: I think that our students probably won't be able to even hypothesize an answer for this as they know they don't know the implications of where state should live -- As a lecturer I'd probably make this a rhetorical question at the moment.

File: React/nested-event-handling.md [hunk 2]
```diff
@@ -1,66 +1,89 @@
...
-        }
-      ];
+      {
+        fullName: "Ada Lovelace",
+        email: "ada@lovelace.uk",
+        isPresent: false
+      },
+      {
+        fullName: "Katherine Johnson",
+        email: "kat@nasa.gov",
```
Comment (tildeee) [inline React/nested-event-handling.md]: v good sample data thank you

File: React/nested-event-handling.md [hunk 3]
```diff
@@ -1,66 +1,89 @@
...
-        },
-        {
-          fullName: "Grade Ada",
-          email: "grace@ada.co",
-          isPresent: false
-        }
-      ];
+      {
+        fullName: "Ada Lovelace",
+        email: "ada@lovelace.uk",
```
Comment (tildeee) [inline React/nested-event-handling.md]: yes good

File: React/prop-types.md [hunk 4]
```diff
@@ -90,7 +86,7 @@ A `defaultProp` attribute like above ensures that every time the given props hav
 You can also create a custom validator which you can look up along with more examples of using `PropTypes` on the [PropTypes github page](https://github.com/facebook/prop-types).
 
 ## Key Takeaway
-`propTypes` is a static attribute you can define in a React component class which lists the props the component can take along with their types.  Mismatches or missing props will generate error messages in the browser console.  PropTypes help you document the props a component can take and help prevent the wrong types of props being passed accidentally.
+`propTypes` is an attribute you can define in a React component which lists the props the component can take along with their types.  Mismatches or missing props will generate error messages in the browser console.  PropTypes help you document the props a component can take and help prevent the wrong types of props being passed accidentally.
```
Comment (tildeee) [inline React/prop-types.md]: So... students won't know what `static` is in terms of JavaScript officially by this point? We have definitions for it in Ruby but I think we still call them class methods and never static, and we have a definition of `static` methods for JavaScript classes on Wednesday (this lesson is slated for Tuesday)... Also, a static method on a class isn't quite the same as what's happening here...

Not sure what to do except maybe have a one sentence explanation somewhere.

Comment (droberts-ada) [reply React/prop-types.md]: Aside from just being static, it's weird b/c we're attaching a property to a function, which they haven't seen before.

Comment (tildeee) [reply React/prop-types.md]: "functions are objects and have properties"?

**Pull request merged**